tests/user_examples/extra_src/shared/porous_media_dynamics.cpp

```diff
@@ -0,0 +1,173 @@
+    Stress_(particles_->Stress_)
+    {
+        numerical_dissipation_factor_ = 0.25;	
+
+        fluid_initial_density = particles_->porous_solid_.getFulidInitialDensity();
+        diffusivity_constant = particles_->porous_solid_.getDiffusivityConstant();
+        water_pressure_constant= particles_->porous_solid_.getWaterPressureConstant();
+    }
+//=================================================================================================//
+void PorousMediaStressRelaxationFirstHalf::initialization(size_t index_i, Real dt)
+{
+    pos_[index_i] += vel_[index_i] * dt * 0.5;
+    F_[index_i] += dF_dt_[index_i] * dt * 0.5;
+    Real J = F_[index_i].determinant();
+    Matd inverse_F_T = F_[index_i].inverse().transpose();
+    Matd  almansi_strain = 0.5 * (Matd::Identity() - (F_[index_i] * F_[index_i].transpose()).inverse());
+    
+    // first, update solid density and volume
+    Vol_update_[index_i] = Vol_[index_i] * J;
+        
+    //obtain the Stress_PK one without J, J is added in total momentum
+    //Note if anisotropic kernel is used, B correction cannot be applied here
+    Stress_[index_i] = (particles_->porous_solid_.StressCauchy(almansi_strain, F_[index_i], index_i)
+        - water_pressure_constant * (fluid_saturation_[index_i] - Eps) * Matd::Identity() ) * inverse_F_T;
+
+    outer_fluid_velocity_relative_fluid_flux_[index_i] = fluid_velocity_[index_i] * relative_fluid_flux_[index_i].transpose() * inverse_F_T;
+          
+}
+//=================================================================================================//
+void PorousMediaStressRelaxationFirstHalf::interaction(size_t index_i, Real dt)
+{
+    Vecd total_momentum_increment = Vecd::Zero();
+    Neighborhood &inner_neighborhood = inner_configuration_[index_i];
+
+    for (size_t n = 0; n != inner_neighborhood.current_size_; ++n)
+    {
+        size_t index_j = inner_neighborhood.j_[n];
+        Real r_ij = inner_neighborhood.r_ij_[n];
+        Vecd gradw_ijV_j_ = inner_neighborhood.dW_ijV_j_[n] * inner_neighborhood.e_ij_[n];
+ 
+        Real dim_r_ij_1 = Dimensions / r_ij;
+        Vecd pos_jump = pos_[index_i] - pos_[index_j];
+        Vecd vel_jump = vel_[index_i] - vel_[index_j];
+        Real strain_rate =  pos_jump.dot(vel_jump) * dim_r_ij_1 * dim_r_ij_1;
+        Real weight = inner_neighborhood.W_ij_[n] * inv_W0_;
+
+        Matd numerical_stress_ij = 0.5 * (F_[index_i] + F_[index_j]) 
+            * particles_->porous_solid_.PairNumericalDamping(strain_rate,  smoothing_length_);
+    
+        //three parts for the momentum increment
+        total_momentum_increment += (Stress_[index_i] + Stress_[index_j])* gradw_ijV_j_
+                                    +  numerical_dissipation_factor_  * numerical_stress_ij * weight * gradw_ijV_j_ ;
+    
+        total_momentum_increment -= (outer_fluid_velocity_relative_fluid_flux_[index_i] + outer_fluid_velocity_relative_fluid_flux_[index_j])
+                                     *gradw_ijV_j_;
+    }
```

Comment:
If is the following better?
total_momentum_increment += (Stress_[index_i] + Stress_[index_j] 
\+  numerical_dissipation_factor_  * numerical_stress_ij * weight 
\- outer_fluid_velocity_relative_fluid_flux_[index_i] - outer_fluid_velocity_relative_fluid_flux_[index_j]) 
\* gradw_ijV_j_;